File: src/domain/types/trade.ts
import type { Mood, TradeType } from './constants'

/** 交易记录 */
export type Trade = {
  id: number
  assetId: number
  planId: number | null
  tradeAt: string
  tradeType: TradeType
  quantity: number
  price: number
  totalAmount: number
  fee: number
  indexPoint: number | null
  reason: string | null
  followPlan: boolean
  mood: Mood | null
  notes: string | null
  createdAt: string
  updatedAt: string
  assetCode?: string | null
  assetName?: string | null
  planStatus?: string | null
  realizedPnl?: number | null
}

/** 创建交易载荷 */
export type TradeCreatePayload = Omit<
  Trade,
  'id' | 'createdAt' | 'updatedAt' | 'assetCode' | 'assetName' | 'planStatus' | 'realizedPnl'
>

/** 更新交易载荷 */
export type TradeUpdatePayload = Omit<
  Trade,
  'createdAt' | 'updatedAt' | 'assetCode' | 'assetName' | 'planStatus' | 'realizedPnl'
>

/** 交易汇总 */
export type TradeSummary = {
  assetId: number
  totalBuyQuantity: number
  totalSellQuantity: number
  currentQuantity: number
  avgCost: number
  remainingCost: number
  realizedPnl: number
  totalBuyAmount: number
  totalSellAmount: number
}

/** 交易过滤条件 */
export type TradeFilter = {
  keyword?: string
  tradeType?: TradeType | ''
  startDate?: string
  endDate?: string
  followPlan?: boolean | ''
  mood?: Mood | ''
}

/** 交易类型标签映射 */
export const TRADE_TYPE_LABELS: Record<TradeType, string> = {
  buy: '买入',
  sell: '卖出',
}

/** 情绪标签映射 */
export const MOOD_LABELS: Record<Mood, string> = {
  calm: '平静',
  anxious: '焦虑',
  greedy: '贪婪',
  fearful: '恐惧',
  hesitant: '犹豫',
  other: '其他',
  unknown: '未知',
}

File: src-tauri/src/models/trade.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Trade {
    pub id: i64,
    pub asset_id: i64,
    pub plan_id: Option<i64>,
    pub trade_at: String,
    pub trade_type: String,
    pub quantity: i64,
    pub price: i64,
    pub total_amount: i64,
    pub fee: i64,
    pub index_point: Option<i64>,
    pub reason: Option<String>,
    pub follow_plan: bool,
    pub mood: Option<String>,
    pub notes: Option<String>,
    pub created_at: String,
    pub updated_at: String,
    pub asset_code: Option<String>,
    pub asset_name: Option<String>,
    pub plan_status: Option<String>,
    pub realized_pnl: Option<i64>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateTradePayload {
    #[serde(alias = "asset_id")]
    pub asset_id: i64,
    #[serde(alias = "plan_id")]
    pub plan_id: Option<i64>,
    #[serde(alias = "trade_at")]
    pub trade_at: String,
    #[serde(alias = "trade_type")]
    pub trade_type: String,
    pub quantity: i64,
    pub price: i64,
    #[serde(alias = "total_amount")]
    pub total_amount: i64,
    pub fee: i64,
    #[serde(alias = "index_point")]
    pub index_point: Option<i64>,
    pub reason: Option<String>,
    #[serde(alias = "follow_plan")]
    pub follow_plan: Option<bool>,
    pub mood: Option<String>,
    pub notes: Option<String>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UpdateTradePayload {
    pub id: i64,
    #[serde(alias = "asset_id")]
    pub asset_id: i64,
    #[serde(alias = "plan_id")]
    pub plan_id: Option<i64>,
    #[serde(alias = "trade_at")]
    pub trade_at: String,
    #[serde(alias = "trade_type")]
    pub trade_type: String,
    pub quantity: i64,
    pub price: i64,
    #[serde(alias = "total_amount")]
    pub total_amount: i64,
    pub fee: i64,
    #[serde(alias = "index_point")]
    pub index_point: Option<i64>,
    pub reason: Option<String>,
    #[serde(alias = "follow_plan")]
    pub follow_plan: bool,
    pub mood: Option<String>,
    pub notes: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TradeSummary {
    pub asset_id: i64,
    pub total_buy_quantity: i64,
    pub total_sell_quantity: i64,
    pub current_quantity: i64,
    pub avg_cost: i64,
    pub remaining_cost: i64,
    pub realized_pnl: i64,
    pub total_buy_amount: i64,
    pub total_sell_amount: i64,
}
